client/shared/record_store/base_records_interface.coffee

```diff
@@ -4,22 +4,26 @@ utils         = require './utils'
 module.exports =
   class BaseRecordsInterface
     model: 'undefinedModel'
+    collectionName: null
+
+    # override this if your apiEndPoint is not the model.plural
+    apiEndPoint: null
 
     constructor: (recordStore) ->
       @baseConstructor recordStore
 
     baseConstructor: (recordStore) ->
       @recordStore = recordStore
-      @collection = @recordStore.db.addCollection(@model.plural, {indices: @model.indices})
+      @collection = @recordStore.db.addCollection(@collectionName or @model.plural, {indices: @model.indices})
       _.each @model.uniqueIndices, (name) =>
         @collection.ensureUniqueIndex(name)
 
-      @remote = new RestfulClient(@model.apiEndPoint or @model.plural)
+      @remote = new RestfulClient(@apiEndPoint or @model.plural)
 
     onInterfaceAdded: ->
 
-    setRemoteCallbacks: (callbacks) ->
-      _.merge @remote, _.pick(callbacks, ['onPrepare', 'onSuccess', 'onUploadSuccess', 'onFailure', 'onCleanup'])
+    setRemoteCallbacks: (callbacks, remote = @remote) ->
```

Comment:
Don't think we need this change anymore (and I'd be hesitant to allow passing in of remotes here, seems dangerous)